### ISSUE TITLE
Progress Bar

### ISSUE THREAD

**Issue by KeyFac** [Brhsoftco/GitHubUpdater#5]
Hi, it's me again.
I wanted to add a form, which includes a progress bar that shows the download progress.
But I couldn't find which part of the code does the download job.
Could you please tell me?
(Machine translation)

**Comment by KeyFac** [comment]
I just found this:
```cs
var responseBytes = ResourceGrab.GrabBytes(downloadJob.DownloadUri);
```
It seems that it uses a method called ```GrabBytes``` that is in ```PlexDL``` , so I questioned this.

**Comment by Brhsoftco** [comment]
Yes, it’s just a byte downloader. I believe this function is located in the PlexDL.Common library?

**Comment by KeyFac** [comment]
It's in PlexDL.AltoHTTP/Common/Net/ResourceGrab.cs
But I still can't understand its logic...
I tried to change ```GrabBytes``` into ```WebClient.DownloadData``` , but I don't know how to change things around this.
It's really bad experience without a progress bar.
(Also I hope it can be async. I mean I can use my app while downloading file.)

**Comment by Brhsoftco** [comment]
Hello, yes PlexDL is synchronous but it operates using a WaitWindow whenever a lengthy task needs to be completed. This essentially puts the work into a background thread and it doesn’t relieve the WaitWindow until this thread closes.

I would suggest utilising the AltoHTTP portion itself, as you can observe how this works using PlexDL’s main form. A progress bar with a download speed estimate is in place there; AltoHTTP is a CodeProject library designed to download using progress updates.
